### PR TITLE
Implement add question submission form

### PR DESCRIPTION
**Pull request by ironfroggy**

### CODE REVIEW

**Comment by dpoirier** [comment]
I see the "Submit a new question link" on the `/` page, but not e.g. `/questions/9/vote/`.  Should it be on all pages?

**Comment by dpoirier** [comment]
To me, counting up to the max length in a text field seems easier to understand than counting down the remaining room.  But maybe that's just me.

**Comment by dpoirier** [comment]
I submitted a question with too much text in some of the fields and ended up on a page that looked like this:

2016-02-12-125445_3280x1200_scrot

Apart from the header, the rest of the page was blank and there was no way to fix the question - no form  displayed.

**Comment by dpoirier** [comment]
Can we make the entry fields not accept more text than we're going to allow?

**Comment by dpoirier** [comment]
The question field is initially displaying "80/200" and counting down 79/200, 78/200, etc...

File: opendebates/templates/opendebates/snippets/add_question.html [hunk 1]
```diff
@@ -20,56 +20,59 @@
...
+            <p class="help-block">{{ form.errors.question }}</p>
+          {% endif %}
           <div class="controls">
-            <textarea id="question" name="question" type="text" rows="4" class="input-large"
+            <textarea id="question" name="question" type="text" rows="4" class="input-large" maxlength="200"
                       placeholder="{% blocktrans %}Enter your question here. If you have multiple&nbsp;questions, please submit each one separately."{% endblocktrans %}>{{ form.data.question }}</textarea>
-            {% if form.errors.question %}
-              <p class="help-block">{{ form.errors.question }}</p>
-            {% endif %}
+            <div class="text-limit-counter" data-total="80"><span class="count">200</span> / 200</div>
```
Comment (dpoirier) [inline opendebates/templates/opendebates/snippets/add_question.html]: data-total="200" here?

**Comment by ironfroggy** [comment]
@dan yes the designs call for this not on the detail pages just list

**Comment by dpoirier** [comment]
On a form submission error, we get to the /questions/ page, which doesn't display any questions, which looks wrong.  I'm not sure if that's the bug here (should /questions/ display some questions like the home page), or if we should just go to / on an error?

**Comment by dpoirier** [comment]
Tried to submit empty form and got traceback in the browser.

Environment:


Request Method: POST
Request URL: http://localhost:8000/questions/

Django Version: 1.8.8
Python Version: 2.7.6
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.sites',
 'django.contrib.flatpages',
 'pipeline',
 'djangobower',
 'dbbackup',
 'djcelery',
 'opendebates',
 'opendebates_emails',
 'djorm_pgfulltext',
 'endless_pagination',
 'bootstrapform',
 'registration',
 'debug_toolbar']
Installed Middleware:
(u'debug_toolbar.middleware.DebugToolbarMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.locale.LocaleMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.flatpages.middleware.FlatpageFallbackMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'opendebates.router.DBRoutingMiddleware')


Traceback:
File "/home/dpoirier/clients/boldprogressives/VE/django-opendebates/local/lib/python2.7/site-packages/django/core/handlers/base.py" in get_response
  223.                 response = middleware_method(request, response)
File "/home/dpoirier/clients/boldprogressives/VE/django-opendebates/local/lib/python2.7/site-packages/django/contrib/messages/middleware.py" in process_response
  23.             unstored_messages = request._messages.update(response)
File "/home/dpoirier/clients/boldprogressives/VE/django-opendebates/local/lib/python2.7/site-packages/django/contrib/messages/storage/base.py" in update
  142.             return self._store(messages, response)
File "/home/dpoirier/clients/boldprogressives/VE/django-opendebates/local/lib/python2.7/site-packages/django/contrib/messages/storage/fallback.py" in _store
  49.                                           remove_oldest=False)
File "/home/dpoirier/clients/boldprogressives/VE/django-opendebates/local/lib/python2.7/site-packages/django/contrib/messages/storage/cookie.py" in _store
  119.         self._update_cookie(encoded_data, response)
File "/home/dpoirier/clients/boldprogressives/VE/django-opendebates/local/lib/python2.7/site-packages/django/contrib/messages/storage/cookie.py" in _update_cookie
  85.             response.set_cookie(self.cookie_name, encoded_data,

Exception Type: AttributeError at /questions/
Exception Value: 'dict' object has no attribute 'set_cookie'

File: opendebates/views.py [hunk 2]
```diff
@@ -255,21 +255,11 @@ def vote(request, id):
     return redirect(url)
 
 
-@rendered_with("opendebates/list_ideas.html")
 @allow_http("GET", "POST")
```
Comment (dpoirier) [inline opendebates/views.py]: The `@rendered_with` is needed if returning a dictionary at line 270 needs to work.

**Comment by ironfroggy** [comment]
updated returning the template for the error page

**Comment by dpoirier** [comment]
If I don't set the category, the error displayed is `"--" is not a valid value for a primary key.` :-)  Is there a way to improve that?  (Not necessarily in this pull request.)

**Comment by dpoirier** [comment]
one comment to fix in this or another pull, then :shipit:

**Pull request merged**